src/docker.ts

```diff
@@ -37,24 +40,53 @@ const saveDockerImages = async (): Promise<void> => {
         "read-only option was selected."
     );
   } else {
-    const preexistingImages = getState(DOCKER_IMAGES_LIST).split("\n");
-    info("Listing Docker images.");
-    const images = await execBashCommand(LIST_COMMAND);
-    const imagesList = images.split("\n");
-    const newImages = imagesList.filter(
-      (image: string): boolean => !preexistingImages.includes(image)
-    );
-    if (newImages.length === 0) {
-      info("No Docker images to save");
+    /* Access the private method, getCacheEntry, internal to the GitHub Actions
+     * cache action.
+     */
+    interface ArtifactCacheEntry {
+      cacheKey?: string;
+      scope?: string;
+      cacheVersion?: string;
+      creationTime?: string;
+      archiveLocation?: string;
+    }
+    interface CacheWithPrivateMethod extends CacheType {
+      getCacheEntry(
+        keys: string[],
+        paths: string[]
+      ): Promise<ArtifactCacheEntry | null>;
+    }
+    const cacheWithPrivateMethod = cache as CacheWithPrivateMethod;
+    const getCacheEntry = cacheWithPrivateMethod["getCacheEntry"];
+    /* Check if a cache with our key has been saved between when we checked in
+     * loadDockerImages and now.
+     */
+    const cacheHit = Boolean(await getCacheEntry([key], [DOCKER_IMAGES_PATH]));
+    if (cacheHit) {
+      info(
+        "A cache miss occurred in loadDockerImages, but there was a subsequent" +
+          "cache hit when we were in saveDockerImages. Not saving cache."
```

Comment:
Consider rewording to clarify what this case corresponds to; the reader should comprehend why the cache was already saved. It's slightly misleading to describe this case as a cache hit since we aren't restoring the cache.

Reply:
Are you wanting me to explicitly mention that this would occur if multiple jobs are run in parallel? Could this also happen if two separate PRs were running through CI using docker cache simultaneously? Are you wanting me to mention these scenarios explicitly? I tentatively changed it to 
```
"A cache miss occurred in loadDockerImages; subsequently a cache with" +
        "a matching key was saved. Not saving cache."
```
but I can mention those specific cases (or others as appropriate) if you'd like me to.

Reply:
Yeah, I think the parallel job case is worth mentioning. It could also occur with parallel workflows like you described, but I don't think we need to go into that level of detail since folks should generally be able to extrapolate from an example. I like the new wording besides that though.